frame/election-provider-multi-phase/src/benchmarking.rs

```diff
@@ -249,7 +317,8 @@ frame_benchmarking::benchmarks! {
 		let encoded_call = <Call<T>>::submit_unsigned(raw_solution.clone(), witness).encode();
 	}: {
 		assert_ok!(<MultiPhase<T>>::submit_unsigned(RawOrigin::None.into(), raw_solution, witness));
-		let _decoded_snap = <RoundSnapshot<T::AccountId> as Decode>::decode(&mut &*encoded_snapshot).unwrap();
+		let _decoded_snap = <RoundSnapshot<T::AccountId> as Decode>::decode(&mut &*encoded_snapshot)
+			.unwrap();
```

Comment:
What is the purpose of decoding here?

Reply:
I believe they are trying to take into account decode time to the benchmarking. Especially if it is a very large and potentially expensive storage item.

Reply:
indeed.

Reply:
Ok I think I see: is it that `feasibility_check`, which is in `submit_unsigned`'s code path, does not return a weight and the most weight intensive part of it is decoding the snapshot?

If this is the case though, shouldn't decoding be taken into account because we have a scenario that will read in the snapshot?